### PR TITLE
FEAT: Add DataFramePlotData.

### PR DESCRIPTION
**Pull request by cfarrow**
This is for you, @jonathanrocher. It needs a bit more testing around 'index' and an example.

### CODE REVIEW

**Comment by jwiggins** [comment]
You should add pandas to the CI requirements

**Comment by cfarrow** [comment]
Thanks for the tip. it took a few tries to get it right. :)

Code + example + tests (passing) = ready for review and merge

File: chaco/data_frame_plot_data.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+# Local, relative imports
+from .abstract_plot_data import AbstractPlotData
+
+
+class DataFramePlotData(AbstractPlotData):
+    """ A PlotData implementation class that handles a DataFrame.
+
+    By default, it doesn't allow its input data to be modified by downstream
+    Chaco components or interactors. The index is availble as data unless there
```
Comment (jvkersch) [inline chaco/data_frame_plot_data.py]: typo "available"

Comment (cfarrow) [inline chaco/data_frame_plot_data.py]: I cannot spell that word right. I need to either stop using it or spell check every time I do. :D

**Comment by cfarrow** [comment]
Any other suggestions?

**Comment by jvkersch** [comment]
Looks great to me! Esp. the demo.

**Comment by cfarrow** [comment]
Thanks!

**Pull request merged**